src/Type/Php/StrSplitFunctionReturnTypeExtension.php

```diff
@@ -67,47 +71,70 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 				$strings = $scope->getType($functionCall->getArgs()[2]->value)->getConstantStrings();
 				$values = array_unique(array_map(static fn (ConstantStringType $encoding): string => $encoding->getValue(), $strings));
 
-				if (count($values) !== 1) {
-					return null;
-				}
-
-				$encoding = $values[0];
-				if (!$this->isSupportedEncoding($encoding)) {
-					return new ConstantBooleanType(false);
+				if (count($values) === 1) {
+					$encoding = $values[0];
+					if (!$this->isSupportedEncoding($encoding)) {
+						return $this->phpVersion->throwsValueErrorForInternalFunctions() ? new NeverType() : new ConstantBooleanType(false);
+					}
 				}
 			} else {
 				$encoding = mb_internal_encoding();
 			}
 		}
 
-		if (!isset($splitLength)) {
-			return null;
-		}
-
 		$stringType = $scope->getType($functionCall->getArgs()[0]->value);
-
-		$constantStrings = $stringType->getConstantStrings();
-		if (count($constantStrings) > 0) {
-			$results = [];
-			foreach ($constantStrings as $constantString) {
-				$items = $encoding === null
-					? str_split($constantString->getValue(), $splitLength)
-					: @mb_str_split($constantString->getValue(), $splitLength, $encoding);
-				if ($items === false) {
-					throw new ShouldNotHappenException();
+		if (
+			isset($splitLength)
+			&& ($functionReflection->getName() === 'str_split' || $encoding !== null)
+		) {
+			$constantStrings = $stringType->getConstantStrings();
+			if (count($constantStrings) > 0) {
+				$results = [];
+				foreach ($constantStrings as $constantString) {
+					$value = $constantString->getValue();
+
+					if ($encoding === null && $value === '') {
```

Comment:
This logic exists to avoid the same issue than https://github.com/phpstan/phpstan/issues/13129